dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/h2/V7.0_2022.04.29__process_to_group.sql

```diff
@@ -0,0 +1,17 @@
+--
+-- The contents of this file are subject to the license and copyright
+-- detailed in the LICENSE and NOTICE files at the root of the source
+-- tree and available online at
+--
+-- http://www.dspace.org/license/
+--
+
+-------------------------------------------------------------------------------
+-- Sequences for Process within Group feature
+-------------------------------------------------------------------------------
+
+CREATE TABLE Process2Group
+(
+    process_id INTEGER REFERENCES Process(process_id),
+    group_id UUID REFERENCES epersongroup (uuid) ON DELETE CASCADE
+);
```

Comment:
Do we need to consider creating a primary key here?  I was comparing this to how we setup similar tables (e.g. `process2bitstream`, `community2collection`, etc.  In all other tables we have a combined primary key specified, as it ensures uniqueness. For example: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/postgres/V7.0_2019_06_14__scripts-and-process.sql#L30-L35

So, I'd expect us to do something similar for this table.  We should have a primary key across `process_id` and `group_id`

Reply:
due to the fact that specialGroups now are a set in Context https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/core/Context.java#L95 we can enforce this constraint as there is no risk that the same group is returned multiple times